p2p/p2p-core/src/client/handshaker/builder.rs

```diff
@@ -0,0 +1,427 @@
+            address_book,
+            core_sync_svc,
+            peer_sync_svc,
+            our_basic_node_data,
+            broadcast_stream_maker,
+            connection_parent_span,
+            _zone,
+            ..
+        } = self;
+
+        HandshakerBuilder {
+            address_book,
+            core_sync_svc,
+            peer_sync_svc,
+            protocol_request_svc: new_protocol_handler,
+            our_basic_node_data,
+            broadcast_stream_maker,
+            connection_parent_span,
+            _zone,
+        }
+    }
+
+    /// Changes the broadcast stream maker, which is used to create streams that yield messages to broadcast.
+    ///
+    /// ## Default Broadcast Stream Maker
+    ///
+    /// The default broadcast stream maker just returns [`stream::Pending`], i.e. the returned stream will not
+    /// produce any messages to broadcast, this is not a problem if your use case does not require broadcasting
+    /// message.
+    pub fn with_broadcast_stream_maker<NBrdcstStrmMkr, BrdcstStrm>(
+        self,
+        new_broadcast_stream_maker: NBrdcstStrmMkr,
+    ) -> HandshakerBuilder<N, AdrBook, CSync, PSync, ProtoHdlr, NBrdcstStrmMkr>
+    where
+        BrdcstStrm: Stream<Item = BroadcastMessage> + Send + 'static,
+        NBrdcstStrmMkr: Fn(InternalPeerID<N::Addr>) -> BrdcstStrm + Clone + Send + 'static,
+    {
+        let HandshakerBuilder {
+            address_book,
+            core_sync_svc,
+            peer_sync_svc,
+            protocol_request_svc,
+            our_basic_node_data,
+            connection_parent_span,
+            _zone,
+            ..
+        } = self;
+
+        HandshakerBuilder {
+            address_book,
+            core_sync_svc,
+            peer_sync_svc,
+            protocol_request_svc,
+            our_basic_node_data,
+            broadcast_stream_maker: new_broadcast_stream_maker,
+            connection_parent_span,
+            _zone,
+        }
+    }
+
+    /// Changes the parent [`Span`] of the connection task to the one provided.
+    ///
+    /// ## Default Broadcast Stream Maker
+    ///
+    /// The default connection span will be [`Span::none`].
+    pub fn with_connection_parent_span(self, connection_parent_span: Span) -> Self {
+        Self {
+            connection_parent_span: Some(connection_parent_span),
+            ..self
+        }
+    }
+
+    /// Builds the [`HandShaker`].
+    pub fn build(self) -> HandShaker<N, AdrBook, CSync, PSync, ProtoHdlr, BrdcstStrmMkr> {
+        HandShaker::new(
+            self.address_book,
+            self.peer_sync_svc,
+            self.core_sync_svc,
+            self.protocol_request_svc,
+            self.broadcast_stream_maker,
+            self.our_basic_node_data,
+            self.connection_parent_span.unwrap_or(Span::none()),
+        )
+    }
+}
+
+/// A dummy peer sync service, that doesn't actually keep track of peers sync states.
+#[derive(Debug, Clone)]
+pub struct DummyPeerSyncSvc;
```

Comment:
These dummy values should be in `p2p/p2p-core/src/client/handshaker/dummy_${THING}.rs` or maybe in a single `dummy.rs` file.